File: src/gatling/scala/org/openapitools/client/model/Interval.scala

package org.openapitools.client.model


case class Interval (
    /* Unique study key */
    _studyKey: Option[String],
    /* Unique system identifier for the interval (visit definition) */
    _intervalId: Option[Integer],
    /* Name of the interval (visit) as defined in the study */
    _intervalName: Option[String],
    /* Description of the interval (visit) */
    _intervalDescription: Option[String],
    /* Sequence number of the interval in the schedule */
    _intervalSequence: Option[Integer],
    /* Identifier for the interval group (if intervals are grouped) */
    _intervalGroupId: Option[Integer],
    /* Name of the interval group */
    _intervalGroupName: Option[String],
    /* Type of interval visit window (e.g., None, Due Date, Start - End Date, Actual Date) */
    _timeline: Option[String],
    /* Baseline interval used for calculating this interval’s dates */
    _definedUsingInterval: Option[String],
    /* Baseline form (name) from which the calculation date is taken */
    _windowCalculationForm: Option[String],
    /* Baseline field (variable name) from which the calculation date is taken */
    _windowCalculationDate: Option[String],
    /* Form used to capture the actual date for this interval */
    _actualDateForm: Option[String],
    /* Field (variable name) used to capture the actual date for this interval */
    _actualDate: Option[String],
    /* Number of days from the calculation date when the interval is due */
    _dueDateWillBeIn: Option[Integer],
    /* Number of days before the due date that are allowed (negative window) */
    _negativeSlack: Option[Integer],
    /* Number of days after the due date that are allowed (positive window) */
    _positiveSlack: Option[Integer],
    /* Number of days of grace period for ePRO completion after due date */
    _eproGracePeriod: Option[Integer],
    /* List of forms that are scheduled in this interval */
    _forms: Option[List[ComponentsSchemasIntervalFormsItem]],
    /* Whether the interval is soft-deleted (disabled) */
    _disabled: Option[Boolean],
    /* Date when this interval was created */
    _dateCreated: Option[String],
    /* Date when this interval was last modified */
    _dateModified: Option[String]
)
object Interval {
    def toStringBody(var_studyKey: Object, var_intervalId: Object, var_intervalName: Object, var_intervalDescription: Object, var_intervalSequence: Object, var_intervalGroupId: Object, var_intervalGroupName: Object, var_timeline: Object, var_definedUsingInterval: Object, var_windowCalculationForm: Object, var_windowCalculationDate: Object, var_actualDateForm: Object, var_actualDate: Object, var_dueDateWillBeIn: Object, var_negativeSlack: Object, var_positiveSlack: Object, var_eproGracePeriod: Object, var_forms: Object, var_disabled: Object, var_dateCreated: Object, var_dateModified: Object) =
        s"""
        | {
        | "studyKey":$var_studyKey,"intervalId":$var_intervalId,"intervalName":$var_intervalName,"intervalDescription":$var_intervalDescription,"intervalSequence":$var_intervalSequence,"intervalGroupId":$var_intervalGroupId,"intervalGroupName":$var_intervalGroupName,"timeline":$var_timeline,"definedUsingInterval":$var_definedUsingInterval,"windowCalculationForm":$var_windowCalculationForm,"windowCalculationDate":$var_windowCalculationDate,"actualDateForm":$var_actualDateForm,"actualDate":$var_actualDate,"dueDateWillBeIn":$var_dueDateWillBeIn,"negativeSlack":$var_negativeSlack,"positiveSlack":$var_positiveSlack,"eproGracePeriod":$var_eproGracePeriod,"forms":$var_forms,"disabled":$var_disabled,"dateCreated":$var_dateCreated,"dateModified":$var_dateModified
        | }
        """.stripMargin
}

File: models/Interval.js
const utils = require('../utils/utils');
const components_schemas_Interval_forms_item = require('../models/components_schemas_Interval_forms_item');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}studyKey`,
                label: `Unique study key - [${labelPrefix}studyKey]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}intervalId`,
                label: `Unique system identifier for the interval (visit definition) - [${labelPrefix}intervalId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}intervalName`,
                label: `Name of the interval (visit) as defined in the study - [${labelPrefix}intervalName]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}intervalDescription`,
                label: `Description of the interval (visit) - [${labelPrefix}intervalDescription]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}intervalSequence`,
                label: `Sequence number of the interval in the schedule - [${labelPrefix}intervalSequence]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}intervalGroupId`,
                label: `Identifier for the interval group (if intervals are grouped) - [${labelPrefix}intervalGroupId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}intervalGroupName`,
                label: `Name of the interval group - [${labelPrefix}intervalGroupName]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}timeline`,
                label: `Type of interval visit window (e.g., None, Due Date, Start - End Date, Actual Date) - [${labelPrefix}timeline]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}definedUsingInterval`,
                label: `Baseline interval used for calculating this interval’s dates - [${labelPrefix}definedUsingInterval]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}windowCalculationForm`,
                label: `Baseline form (name) from which the calculation date is taken - [${labelPrefix}windowCalculationForm]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}windowCalculationDate`,
                label: `Baseline field (variable name) from which the calculation date is taken - [${labelPrefix}windowCalculationDate]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}actualDateForm`,
                label: `Form used to capture the actual date for this interval - [${labelPrefix}actualDateForm]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}actualDate`,
                label: `Field (variable name) used to capture the actual date for this interval - [${labelPrefix}actualDate]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}dueDateWillBeIn`,
                label: `Number of days from the calculation date when the interval is due - [${labelPrefix}dueDateWillBeIn]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}negativeSlack`,
                label: `Number of days before the due date that are allowed (negative window) - [${labelPrefix}negativeSlack]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}positiveSlack`,
                label: `Number of days after the due date that are allowed (positive window) - [${labelPrefix}positiveSlack]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}eproGracePeriod`,
                label: `Number of days of grace period for ePRO completion after due date - [${labelPrefix}eproGracePeriod]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}forms`,
                label: `[${labelPrefix}forms]`,
                children: components_schemas_Interval_forms_item.fields(`${keyPrefix}forms${!isInput ? '[]' : ''}`, isInput, true), 
            },
            {
                key: `${keyPrefix}disabled`,
                label: `Whether the interval is soft-deleted (disabled) - [${labelPrefix}disabled]`,
                type: 'boolean',
            },
            {
                key: `${keyPrefix}dateCreated`,
                label: `Date when this interval was created - [${labelPrefix}dateCreated]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}dateModified`,
                label: `Date when this interval was last modified - [${labelPrefix}dateModified]`,
                type: 'string',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'studyKey': bundle.inputData?.[`${keyPrefix}studyKey`],
            'intervalId': bundle.inputData?.[`${keyPrefix}intervalId`],
            'intervalName': bundle.inputData?.[`${keyPrefix}intervalName`],
            'intervalDescription': bundle.inputData?.[`${keyPrefix}intervalDescription`],
            'intervalSequence': bundle.inputData?.[`${keyPrefix}intervalSequence`],
            'intervalGroupId': bundle.inputData?.[`${keyPrefix}intervalGroupId`],
            'intervalGroupName': bundle.inputData?.[`${keyPrefix}intervalGroupName`],
            'timeline': bundle.inputData?.[`${keyPrefix}timeline`],
            'definedUsingInterval': bundle.inputData?.[`${keyPrefix}definedUsingInterval`],
            'windowCalculationForm': bundle.inputData?.[`${keyPrefix}windowCalculationForm`],
            'windowCalculationDate': bundle.inputData?.[`${keyPrefix}windowCalculationDate`],
            'actualDateForm': bundle.inputData?.[`${keyPrefix}actualDateForm`],
            'actualDate': bundle.inputData?.[`${keyPrefix}actualDate`],
            'dueDateWillBeIn': bundle.inputData?.[`${keyPrefix}dueDateWillBeIn`],
            'negativeSlack': bundle.inputData?.[`${keyPrefix}negativeSlack`],
            'positiveSlack': bundle.inputData?.[`${keyPrefix}positiveSlack`],
            'eproGracePeriod': bundle.inputData?.[`${keyPrefix}eproGracePeriod`],
            'forms': utils.childMapping(bundle.inputData?.[`${keyPrefix}forms`], `${keyPrefix}forms`, components_schemas_Interval_forms_item),
            'disabled': bundle.inputData?.[`${keyPrefix}disabled`],
            'dateCreated': bundle.inputData?.[`${keyPrefix}dateCreated`],
            'dateModified': bundle.inputData?.[`${keyPrefix}dateModified`],
        }
    },
}
